### PR TITLE
Enable keyboard shortcuts for autocomplete @username selection in chat

### LINKED ISSUE #9291: @Username selection with keyboard arrow keys in drop-down mentions list in chat
When tagging someone's name in chat, you can no longer select the person's name using arrow keys. Instead, you need to use the mouse. You should be able to use arrow keys to select someone's name.

### PR DESCRIPTION
**Pull request by jackcogdill**
[//]: # (Note: See http://habitica.fandom.com/wiki/Using_Your_Local_Install_to_Modify_Habitica%27s_Website_and_API for more info)

[//]: # (Put Issue # here, if applicable. This will automatically close the issue if your PR is merged in)
Fixes #9291 

### Changes
[//]: # (Describe the changes that were made in detail here. Include pictures if necessary)

- Enable keyboard shortcuts for chat autocomplete
  - Tab/down to select next
  - Shift-tab/up to select previous
  - Enter to make selection
  - Escape to cancel
- Chat autocomplete mouse hover styling: use cursor pointer to feel more clickable.

[//]: # (Put User ID in here - found on the Habitica website at User Icon > Settings > API)

----
UUID: 5ffee717-5618-4429-b161-981d79842f8a

### CODE REVIEW

**Comment by paglias** [comment]
Hi @jackcogdill and thanks for the PR! Sorry for the delay, we'll review it this week

File: website/client/components/chat/autoComplete.vue [hunk 1]
```diff
@@ -3,8 +3,8 @@
   .autocomplete-results.d-flex.align-items-center(
     v-for='result in searchResults',
     @click='select(result)',
-    @mouseenter='result.hover = true',
-    @mouseleave='result.hover = false',
+    @mouseenter='resetSelection(); result.hover = true',
```
Comment (paglias) [inline website/client/components/chat/autoComplete.vue]: could `result.hover = true` be moved inside `resetSelection `?

Comment (jackcogdill) [reply website/client/components/chat/autoComplete.vue]: No. `resetSelection()` sets `hover` to `false` for all the results, so it wouldn't make sense.

Comment (jackcogdill) [reply website/client/components/chat/autoComplete.vue]: I suppose it's possible to pass result into the function as an optional parameter, and then set it inside the function. Would you prefer that?

Comment (paglias) [reply website/client/components/chat/autoComplete.vue]: yeah it would be better, we try to keep the js into templates to a minimum, thanks!

Comment (jackcogdill) [reply website/client/components/chat/autoComplete.vue]: @paglias I updated the code. Please take another look

**Comment by paglias** [comment]
@jackcogdill thanks for the PR, this seems to be working very well and the code is fine as well. We're going to test this internally for about a week and if not issues arise it'll go live soon, I'll let you know if there's any problem

**Comment by paglias** [comment]
@jackcogdill  I'm merging the PR and awarding your contributor tier.

Note that further tiers require increasing amounts of work from one to the next, so Tier 2 may require a couple of PRs or a larger PR to attain. But keep helping out and we'll express our gratitude accordingly!

**Pull request merged**